src/main/java/de/hpi/bpt/argos/storage/PersistenceAdapterImpl.java

```diff
@@ -0,0 +1,688 @@
+		Transaction transaction = session.beginTransaction();
+
+		Table entityTypeTable = Entity.class.getAnnotation(Table.class);
+		String entityTypeTableName = "EntityType";
+
+		if (entityTypeTable != null) {
+			entityTypeTableName = entityTypeTable.name();
+		}
+
+		String sqlQuery = String.format("SELECT count(*) FROM %1$s", entityTypeTableName);
+		Query query = session.createNativeQuery(sqlQuery);
+
+		String stringResult = databaseAccess.getArtifacts(session, query, transaction, query::getSingleResult, 0).toString();
+		return Integer.parseInt(stringResult);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Event> getEvents(long eventTypeId, int listStartIndex, int listEndIndex, Long... entityIds) {
+		if (entityIds.length == 0) {
+			return new ArrayList<>();
+		}
+
+		StringBuilder sqlWhere = new StringBuilder();
+		sqlWhere.append(String.format("(event.entityId = %1$d)", entityIds[0]));
+
+		for (int i = 1; i < entityIds.length; i++) {
+			sqlWhere.append(String.format(" OR (event.entityId = %1$d)", entityIds[i]));
+		}
+
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		String sqlQuery = String.format(
+				"FROM EventImpl event "
+						+ "WHERE event.typeId = %1$d"
+						+ "AND (%2$s)",
+				eventTypeId,
+				sqlWhere.toString()
+		);
+
+		Query<Event> query = session.createQuery(sqlQuery,
+				Event.class)
+				.setFirstResult(listStartIndex)
+				.setMaxResults(listEndIndex);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::getResultList, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public List<Event> getEvents(long entityOwnerId) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Query<Event> query = session.createQuery("FROM EventImpl event "
+						+ "WHERE event.entityId = :entityOwnerId",
+				Event.class)
+				.setParameter("entityOwnerId", entityOwnerId);
+
+		return databaseAccess.getArtifacts(session, query, transaction, query::list, new ArrayList<>());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 * @param entityIds
+	 */
+	@Override
+	public List<EventType> getEventTypes(Long... entityIds) {
+		if (entityIds.length == 0) {
+			return new ArrayList<>();
+		}
+
+		List<Long> eventTypeIds = getEventTypeIds(entityIds);
+		Session session = databaseAccess.getSessionFactory().openSession();
+		return databaseAccess.getArtifactsById(session, EventTypeImpl.class, eventTypeIds.toArray(new Long[eventTypeIds.size()]));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public int getEventCountOfEntity(long entityId, long eventTypeId) {
+		Session session = databaseAccess.getSessionFactory().openSession();
+		Transaction transaction = session.beginTransaction();
+
+		Table eventTable = EventImpl.class.getAnnotation(Table.class);
+		String eventTableName = "Event";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Event" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)